Assets/Code/src/Runtime/Match/Components/MatchManager.cs

```diff
@@ -72,13 +69,16 @@ public class MatchManager : MonoBehaviour {
       MatchConfig = Config,
       MatchState = MatchController.CurrentState
     });
-    return result;
+	 MatchController.CurrentState.StateID = MatchStateID.End;
+	 return result;
   }
 
   public void SetPaused(bool paused) {
     if (!IsLocal) return;
-    bool changed = paused != IsPaused;
-    IsPaused = paused;
+
+	 var pauseID = paused ? MatchStateID.Pause : MatchStateID.InGame;
```

Comment:
Needs an additional check: you can only pause if StateID is InGame, if it's not Paused or InGame, you shouldn't be doing either.